drivers/serial/uart_stm32.c

```diff
@@ -714,6 +718,8 @@ static int uart_stm32_init(const struct device *dev)
 				LL_GPIO_AF_EnableRemap_USART3();
 			} else if (remap == REMAP_1) {
 				LL_GPIO_AF_RemapPartial_USART3();
+			} else {
+				LL_GPIO_AF_DisableRemap_USART3();
```

Comment:
PWM checks `if (remap == NO_REMAP)` so I'd agree on using the same strategy to keep STM32 drivers consistent, whatever is chosen.

Reply:
Indeed, I hesitated and ended up with this solution.
Reason to select this one is that there is already a function that checks remap setting before arriving here, which limits the possibility of misconfiguration.
On the other hand, to be coherent, checking vs NO_REMAP would deserve adding a final `else` to detect errors.
So ended up with this as it saves few LOC.